exporter-helper.php

```diff
@@ -78,34 +78,64 @@ function get_campaign_attachments( $post_ids ) {
 	foreach ( (array) $results as $text ) {
 		$text = $text->post_content;
 
-		// Filter attachment ids from caption.
-		preg_match_all( '#(wp-image-|wp-att-|attachment\_)(\d+)#', $text, $matches, PREG_SET_ORDER );
-		foreach ( $matches as $match ) {
-			$attachment_ids[] = $match[2];
-		}
+		$blocks = parse_blocks( $text );
+		foreach ( $blocks as $block ) {
+			// Fetch background image id from EN form block.
+			if ( 'planet4-blocks/enform' === $block['blockName'] && isset( $block['attrs']['background'] ) ) {
+				$attachment_ids[] = $block['attrs']['background'];
+			}
 
-		// Filter attachment ids from shortcake code(shortcake_gallery, shortcake_happy_point, shortcake_media_video).
-		preg_match_all( '#wp\:planet4\-blocks\/[a-zA-Z0-9\_\"\'\-\s\:\/\/\=\.\?\&\,\_\{\%]*[\"|\'](multiple_image|background|id|video_poster_img)[\"|\'][\:][\"|\']?([\d\s\,]*)[\"|\']?#', $text, $matches, PREG_SET_ORDER );
-		foreach ( $matches as $match ) {
-			if ( 'multiple_image' === $match[1] ) {
-				$multiple_images = explode( ',', $match[2] );
+			// Fetch HP background image id from HP block.
+			if ( 'planet4-blocks/happypoint' === $block['blockName'] && isset( $block['attrs']['id'] ) ) {
+				$attachment_ids[] = $block['attrs']['id'];
+			}
+
+			// Fetch video poster image id from Media block.
+			if ( 'planet4-blocks/media-video' === $block['blockName'] && isset( $block['attrs']['video_poster_img'] ) ) {
+				$attachment_ids[] = $block['attrs']['video_poster_img'];
+			}
+
+			// Fetch image ids from Gallery block.
+			if ( 'planet4-blocks/gallery' === $block['blockName'] && isset( $block['attrs']['multiple_image'] ) ) {
+				$multiple_images = explode( ',', $block['attrs']['multiple_image'] );
 				$attachment_ids  = array_merge( $attachment_ids, $multiple_images );
-			} else {
-				$attachment_ids[] = $match[2];
 			}
-		}
 
-		// Filter attachment ids from shortcake code(shortcake_carousel_header, shortcake_split_two_columns, shortcake_columns).
-		preg_match_all( '#[\"|\'](image|attachment|issue_image|tag_image)[\"|\'][\:][\"|\']?([\d]*)[\'|\"]?#', $text, $matches, PREG_SET_ORDER );
-		foreach ( $matches as $match ) {
-			$attachment_ids[] = $match[2];
-		}
+			// Fetch image id from Carousel Header block.
+			if ( 'planet4-blocks/carousel-header' === $block['blockName'] && isset( $block['attrs']['slides'] ) ) {
+				foreach ( $block['attrs']['slides'] as $slide ) {
+					$attachment_ids[] = $slide['image'];
+				}
+			}
+
+			// Fetch issue/tag image id from Split Two Columns block.
+			if ( 'planet4-blocks/split-two-columns' === $block['blockName'] && ( isset( $block['attrs']['issue_image'] ) || isset( $block['attrs']['tag_image'] ) ) ) {
+				$attachment_ids[] = $block['attrs']['issue_image'];
```

Comment:
You probably need to use the null coalescing operator here to avoid notices, or you could split it in two if-statements (which is a bit easier when you use a switch for the block name.